crates/optimism/rpc/src/historical_proofs.rs

```diff
@@ -0,0 +1,104 @@
+//! Historical proofs RPC server implementation.
+
+use alloy_eips::BlockId;
+use alloy_primitives::Address;
+use alloy_rpc_types_eth::EIP1186AccountProofResponse;
+use alloy_serde::JsonStorageKey;
+use async_trait::async_trait;
+use derive_more::Constructor;
+use jsonrpsee::proc_macros::rpc;
+use jsonrpsee_core::RpcResult;
+use reth_optimism_trie::{provider::OpProofsStateProviderRef, OpProofsStorage};
+use reth_provider::{BlockIdReader, ProviderError, ProviderResult, StateProviderBox};
+use reth_rpc_api::eth::helpers::FullEthApi;
+use reth_rpc_eth_types::EthApiError;
+
+#[cfg_attr(not(test), rpc(server, namespace = "eth"))]
+#[cfg_attr(test, rpc(server, client, namespace = "eth"))]
+pub trait EthApiOverride {
+    /// Returns the account and storage values of the specified account including the Merkle-proof.
+    /// This call can be used to verify that the data you are pulling from is not tampered with.
+    #[method(name = "getProof")]
+    async fn get_proof(
+        &self,
+        address: Address,
+        keys: Vec<JsonStorageKey>,
+        block_number: Option<BlockId>,
+    ) -> RpcResult<EIP1186AccountProofResponse>;
+}
+
+#[derive(Debug, Constructor)]
+/// Overrides applied to the `eth_` namespace of the RPC API for historical proofs ExEx.
+pub struct EthApiExt<Eth, P> {
+    eth_api: Eth,
+    preimage_store: P,
+}
+
+impl<Eth, P> EthApiExt<Eth, P>
+where
+    Eth: FullEthApi + Send + Sync + 'static,
+    jsonrpsee_types::error::ErrorObject<'static>: From<Eth::Error>,
```

Comment:
```suggestion
    ErrorObject<'static>: From<Eth::Error>,
```
pls import in module prelude